micro-rdk-installer/src/main.rs

```diff
@@ -25,13 +25,26 @@ struct AppConfig {
 
 #[derive(Subcommand)]
 enum Commands {
-    WriteBinary(WriteBinary),
+    WriteCredentials(WriteCredentials),
     WriteFlash(WriteFlash),
     CreateNvsPartition(CreateNVSPartition),
 }
 
 #[derive(Args)]
-struct WriteBinary {}
+struct WriteCredentials {
+    #[arg(long = "app-config")]
+    config: String,
+    #[arg(long = "bin")]
+    binary_path: String,
+    #[arg(long = "nvs-size", default_value = "32768")]
+    size: usize,
+    // the default here represents the offset as declared in
+    // examples/esp32/partitions.csv (0x9000, here written as 36864),
+    // as that is the partition table that will be used to compile
+    // the default application binary
+    #[arg(long = "nvs-offset-address", default_value = "36864")]
+    nvs_offset: u64,
```

Comment:
maybe as a later ticket. But would be nice to search for an nvs slot in the partition table and pull the numbers from there

Reply:
https://viam.atlassian.net/browse/RSDK-4711